### PR TITLE
Form fields only update if props change

### PR DESCRIPTION
**Pull request by kennyt**
This improves performance in huge forms by a large degree

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @MatApple

File: src/Form/FieldCheckbox.js [hunk 1]
```diff
@@ -7,6 +7,9 @@ import IconCheckbox from './icons/IconCheckbox';
 import Util from '../Util/Util';
 
 class FieldCheckbox extends FieldRadioButton {
+  shouldComponentUpdate(nextProps) {
```
Comment (mlunoe) [inline src/Form/FieldCheckbox.js]: This extends FieldRadioButton, so if that implements it, this doesn't have to

**Comment by mlunoe** [comment]
One comment and then :shipit: !

File: src/Form/FieldCheckbox.js (hunk 1 above)
Comment (kennyt) [inline src/Form/FieldCheckbox.js]: done

**Pull request merged**